### PR TITLE
`sgx-detect` support for confidentiality-only protection EPC

### LINKED ISSUE #332: sgx-detect: does not understand EPC information on ice lake
On an ice lake machine:

```
$ ./sgx-detect  --verbose
Detecting SGX, this may take a minute...
？ SGX instruction set
  ✔  CPU support
  ✔  CPU configuration
  ✔  Enclave attributes
  ？ Enclave Page Cache
  SGX features
    ✔  SGX2  ✔  EXINFO  ✘  ENCLV  ✘  OVERSUB  ✔  KSS  
    Total EPC size: 0.0MiB
✔  Flexible launch control
  ✔  CPU support
  ？ CPU configuration
  ✔  Able to launch production mode enclave
✔  SGX system software
  ✔  SGX kernel device (/dev/sgx/enclave)
  ✔  libsgx_enclave_common
  ✔  AESM service
  ✔  Able to launch enclaves
    ✔  Debug mode
    ✔  Production mode
    ✔  Production mode (Intel whitelisted)

🕮  SGX instruction set > Enclave Page Cache
Your hardware supports SGX, but the Enclave Page Cache (EPC) is misconfigured. This could indicate a CPU or hypervisor bug.

debug: CPUID.(EAX=12H,ECX=2H):ECX[3:0] = Unknown
debug: CPUID.(EAX=12H,ECX=3H):ECX[3:0] = Unknown

More information: https://edp.fortanix.com/docs/installation/help/#invalid-cpuid
```

### PR DESCRIPTION
**Pull request by ssavvides**
Fixes #332

### CODE REVIEW

File: sgxs-tools/src/sgx_detect/tests/mod.rs [hunk 1]
```diff
@@ -219,6 +219,11 @@ impl Dependency<SgxCpuSupport> for EnclavePageCache {
                             phys_size,
                             ..
                         } => total_size += phys_size,
+                        Cpuid12hEnum::Epc {
```
Comment (ssavvides) [inline sgxs-tools/src/sgx_detect/tests/mod.rs:222]: avoiding `or_patterns` here

Comment (jethrogb) [reply sgxs-tools/src/sgx_detect/tests/mod.rs:222]: That's stable now

**Review by jethrogb — COMMENTED** [review]
It would be good to print the EPC type now that there's more than one type.

**Comment by jethrogb** [comment]
Please put the formatting changes in a separate commit

File: sgxs-tools/src/sgx_detect/tests/mod.rs [hunk 2]
```diff
@@ -336,14 +352,28 @@ impl Print for EpcSize {
...
-                "{:width$}{}: {:.1}MiB",
-                "",
-                self.name(),
-                epc.total_size as f64 / (1048576.),
-                width = level * 2
-            );
+            print!("{:width$}{}:", "", self.name(), width = level * 2);
+            if epc.total_size_cip > 0 {
+                let (epc_size, epc_unit) = epc_size_unit(epc.total_size_cip);
+                print!(" {:.1}{} (Confidentiality and Integrity Protected)", epc_size, epc_unit);
```
Comment (jethrogb) [inline sgxs-tools/src/sgx_detect/tests/mod.rs:369]: The terms Intel use are unclear, I don't think we should reuse them in our output. I'm still clarifying with Intel what future types will look like. Let's use something like this:

* regular SGX:
```
EPC size: 50.0MiB
```

* Ice lake server:
```
EPC size: 5.0GiB (no integrity protection)
```

File: sgxs-tools/src/sgx_detect/tests/mod.rs [hunk 3]
```diff
@@ -336,14 +352,28 @@ impl Print for EpcSize {
     }
 
     fn print(&self, level: usize) {
+        fn epc_size_unit(total_size: u64) -> (f64, String) {
```
Comment (jethrogb) [inline sgxs-tools/src/sgx_detect/tests/mod.rs:355]: Could've just returned `&'static str` instead of `String`

Comment (ssavvides) [reply sgxs-tools/src/sgx_detect/tests/mod.rs]: Adjusted, please restart bors if needed

**Pull request merged**